src/H5system.c

```diff
@@ -1350,3 +1350,42 @@ H5_get_option(int argc, const char *const *argv, const char *opts, const struct
     /* return the current flag character found */
     return optchar;
 }
+
+/*-------------------------------------------------------------------------
+ * Function:    H5strcasestr
+ *
+ * Purpose:     Find the first occurrence of the substring needle in the
+ *              string haystack ignoring case.
+ *
+ * Return:      Success:  Pointer to the beginning of the located substring
+ *
+ *              Failure:  NULL
+ *
+ * Programmer:  Frank Berghaus
+ *              Sep 2022
+ *
+ *-------------------------------------------------------------------------
+ */
+char *
+H5strcasestr(const char *haystack, const char *needle)
+{
```

Comment:
@berghaus Looks like this function will need to be updated with FUNC_ENTER and FUNC_LEAVE. In this case, it's probably fine to use FUNC_ENTER_NOAPI_NOINIT. You can refer to H5_basename and H5_dirname in this same file for an example how to instrument the function.

Reply:
I don't think we need to use FUNC_ENTER macros here. Most of the replacement functions don't use them and I plan to rip out the name checks in the future anyway.